library/core/src/primitive_docs.rs

```diff
@@ -1082,9 +1082,12 @@ impl<T> (T,) {}
 /// bits. Please see [the documentation for [`prim@f32`] or [Wikipedia on
 /// half-precision values][wikipedia] for more information.
 ///
-/// Note that most major platforms will provide `f16` math support by converting to and from
-/// an `f32`, which is usually fairly performant but will not be as fast as using `f32` directly.
-
+/// Note that most common platforms will not support `f16` in hardware without enabling extra target
+/// features, with the notable exception of Apple Silicon (also known as M1/2/etc) processors. For
```

Comment:
We can spare a few bytes.
```suggestion
/// features, with the notable exception of Apple Silicon (also known as M1, M2, etc.) processors.
```